inference-engine/tests/functional/plugin/cpu/single_layer_tests/one_hot.cpp

```diff
@@ -3,165 +3,313 @@
+        InferenceEngine::Precision outPrc;
         CPUSpecificParams cpuParams;
-        std::tie(inputShape, axis, depth, onValue, offValue, netPrecision, inPrc, outPrc, targetDevice, cpuParams) = obj.param;
+        std::tie(inputShape, axis, inputType, depth, onValue, offValue, outPrc, cpuParams) = obj.param;
 
         std::ostringstream result;
-        result << "IS=" << CommonTestUtils::vec2str(inputShape) << "_";
+        if (inputShape.first.size() != 0) {
+            result << "IS=(" << CommonTestUtils::partialShape2str({inputShape.first}) << "_";
+        }
+        result << "TS=";
+        for (const auto& shape : inputShape.second) {
+                result << CommonTestUtils::vec2str(shape) << "_";
+        }
         result << "axis=" << axis << "_";
-        result << "depth=" << depth << "_";
+        if (inputType.first == ngraph::helpers::InputLayerType::CONSTANT)
+            result << "depth=" << depth << "_";
+        else
+            result << "depth=PARAMETER" << "_";
         result << "OnVal=" << onValue << "_";
         result << "OffVal=" << offValue << "_";
-        result << "netPRC=" << netPrecision.name() << "_";
-        result << "inPRC=" << inPrc.name() << "_";
-        result << "outPRC=" << outPrc.name() << "_";
-        result << "trgDev=" << targetDevice;
+        result << "outPRC=" << outPrc.name();
         result << CPUTestsBase::getTestCaseName(cpuParams);
         return result.str();
     }
+    void generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+        for (int i = 0; i < funcInputs.size(); ++i) {
+            const auto& funcInput = funcInputs[i];
+            ov::runtime::Tensor tensor;
+
+            if (i == 1) {
+                tensor = ov::runtime::Tensor(funcInput.get_element_type(), targetInputStaticShapes[i]);
+                auto *dataPtr = tensor.data<int32_t>();
+                dataPtr[0] = Depth;
+            } else {
+                tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i]);
+            }
+
+            inputs.insert({funcInput.get_node_shared_ptr(), tensor});
+        }
+    }
 protected:
     void SetUp() override {
-        ngraph::Shape inputShape;
-        int axis;
-        size_t depth;
-        float onValue, offValue;
-        InferenceEngine::Precision netPrecision;
-        CPUSpecificParams cpuParams;
+        targetDevice = CommonTestUtils::DEVICE_CPU;
 
-        std::tie(inputShape, axis, depth, onValue, offValue, netPrecision, inPrc, outPrc, targetDevice, cpuParams) = this->GetParam();
+        InputShape inputShape;
+        std::pair<ngraph::helpers::InputLayerType, bool> inputType;
+        InferenceEngine::Precision outPrc;
+        CPUSpecificParams cpuParams;
+        std::tie(inputShape, Axis, inputType, Depth, OnValue, OffValue, outPrc, cpuParams) = this->GetParam();
```

Comment:
if `inputType.second == true` you should generate `Depth` independent on `depth` input type